plugins/com.google.cloud.tools.eclipse.appengine.libraries.ui/src/com/google/cloud/tools/eclipse/appengine/libraries/ui/CloudLibrariesPage.java

```diff
@@ -103,13 +103,15 @@ public void initialize(IJavaProject project, IClasspathEntry[] currentEntries) {
       return null;
     }
 
+    // todo how do we get a real monitor here?
+    NullProgressMonitor monitor = new NullProgressMonitor();
     try {
       if (MavenUtils.hasMavenNature(project.getProject())) {
-        BuildPath.addMavenLibraries(project.getProject(), libraries, new NullProgressMonitor());
+        BuildPath.addMavenLibraries(project.getProject(), libraries, monitor);
         return new IClasspathEntry[0];
       } else {
-        Library masterLibrary = BuildPath.collectLibraryFiles(project, libraries);
-        IClasspathEntry[] added = BuildPath.listNativeLibrary(project, masterLibrary);
+        Library masterLibrary = BuildPath.collectLibraryFiles(project, libraries, monitor);
+        IClasspathEntry[] added = BuildPath.listNativeLibrary(project, masterLibrary, monitor);
```

Comment:
and ditto here with `monitor.newChild(xxx)`

Reply:
done